cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,54 @@
-/// <reference types='cypress' />
+/// <reference types="cypress" />
 
-describe('Student Registration page', () => {
-  before(() => {
+describe('Automation Practice Form', () => {
+  beforeEach(() => {
 
+    cy.on('uncaught:exception', (err, runnable) => {
+      return false;
+    });
+
+
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
 
-  it('', () => {
+  it('should fill out the form and assert data in modal', () => {
+
+    cy.get('#firstName').type('John');
+    cy.get('#lastName').type('Doe');
+    cy.get('#userEmail').type('john.doe@example.com');
+    cy.get('label[for="gender-radio-1"]').click();
+    cy.get('#userNumber').type('1234567890');
+
+
+    cy.get('#dateOfBirthInput').click();
+    cy.get('.react-datepicker__month-select').select('April');
+    cy.get('.react-datepicker__year-select').select('1990');
+    cy.get('.react-datepicker__day--010').select('10');
+
+
+    cy.get('#currentAddress').type('123 Main St, Anytown, USA');
+
+
+    cy.get('.subjects-auto-complete__value-container').type('Math{enter}');
+
+
+    cy.get('input[type="checkbox"][value="Sports"]').check();
```

Comment:
The `check()` method is not applicable for input elements of type checkbox. Use `.click()` instead to select the checkbox.